client/src/ReferralLinkGenerator.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { Card, Button } from 'semantic-ui-react'
+import { toast } from 'react-semantic-toasts'
+
+/**
+ * Button group that allows a user to copy a referral link to share
+ * @param {*} props
+ * schoolId - id for the user current school
+ * schoolName - name of the user school
+ * userId - userId for the student / alumnus 
+ */
+export default function ReferralLinkGenerator (props) {
+
+    const copyLink = (role) => {
+        let referralLink = `${window.location.href}register/${role}/${props.userId}/${props.schoolId}`
+        navigator.clipboard.writeText(referralLink).then(() => {
+            toast(
+                {
+                    title: 'Referral Link Copied',
+                    description: <p>Your {role.toUpperCase()} referral link has been copied to clipboard!</p>,
+                    color: role === 'alumni' ? 'orange' : 'yellow',
+                    time: 2000,
+                    icon: 'paper plane',
+                }
+            );
+        })
+    }
+
+    return (
```

Comment:
also do not quite have developed thoughts on this yet, but the card is not quite distinctive. When reviewing the change, I did not immediately recognise where the refer button was. Things that may help:
* Maybe inverting the colours? And making the buttons solid coloured. That way the they are more apparent and do not blend it with the app background.
* Inversing the order s.t. it is button above, descriptive text below. Because there is no title in the card, I think it would help with app flow if the first thing the user saw was the action instead of the description of the action. If more context is needed then the user will go on to read the description.